### DISCUSSION TITLE
useErrors

### DISCUSSION THREAD

**Discussion by TkDodo** [TanStack/query#1634] [Ideas]
What would we think about a dedicated `useErrors` hook to be able to display errors globally, similar to `useIsFetching`?
instead of the number of fetching queries, we could return a map of queryKey -> error for all queries that are in error state (+ the usual filter options).

we’ve had this use case, and creating subscriptions with `useQuery` just to display errors was cumbersome because building the keys involved subscribing to redux (where part of the queryKey lives). A global hook would solve this nicely - `useErrorBoundary` would be the other alternative, so we have options if this idea is not gonna make it :)

**Comment by waptik** [comment]
Sounds like a great idea. It'll be great having especially in cases where you want to display an error message or do something else in components that depends on a particular query state.
It'll also help in avoiding passing `error` and `isError` to lower level components which can be tedious.

I don't know if anyone gets my point here!
